server/src/main/resources/db/migration/V1_13__Namespace_Name_Upper.sql

```diff
@@ -0,0 +1,4 @@
+-- drop existing case-sensitive unique constraint
+ALTER TABLE namespace DROP CONSTRAINT ukeq2y9mghytirkcofquanv5frf;
+-- create case-insensitive unique index
+CREATE UNIQUE INDEX ukeq2y9mghytirkcofquanv5frf ON namespace(UPPER(name));
```

Comment:
why not lower? would there be a difference from client perspective?

Reply:
@akosyakov I'm optimizing for https://github.com/eclipse/openvsx/blob/08c60ac7d7431cf46acd8e4f7ca0336b4163575c/server/src/main/java/org/eclipse/openvsx/repositories/NamespaceRepository.java#L19

Spring translates this method signature to:
`SELECT * FROM namespace WHERE UPPER(name) = UPPER('foo')`
The UPPER function is used because IgnoreCase is part of the method signature.

There already was a unique constraint on the name column (ukeq2y9mghytirkcofquanv5frf).
However, this index is not used by the database when the upper function is applied. 
This results in a sequential (row by row) scan instead of an index scan.

By using the new unique index the database can perform an index scan, which results in better performance.

Reply:
On second glance, more queries could benefit from this change:
https://github.com/eclipse/openvsx/blob/08c60ac7d7431cf46acd8e4f7ca0336b4163575c/server/src/main/java/org/eclipse/openvsx/repositories/ExtensionRepository.java#L29

https://github.com/eclipse/openvsx/blob/08c60ac7d7431cf46acd8e4f7ca0336b4163575c/server/src/main/java/org/eclipse/openvsx/repositories/ExtensionRepository.java#L37

https://github.com/eclipse/openvsx/blob/08c60ac7d7431cf46acd8e4f7ca0336b4163575c/server/src/main/java/org/eclipse/openvsx/repositories/ExtensionVersionRepository.java#L32

It's similar to extension_namespace_and_name_idx:
https://github.com/eclipse/openvsx/blob/08c60ac7d7431cf46acd8e4f7ca0336b4163575c/server/src/main/resources/db/migration/V1_7__Indexes.sql#L1